tesseract_time_parameterization/src/time_optimal_trajectory_generation.cpp

```diff
@@ -1093,137 +1090,105 @@ bool Trajectory::isValid() const { return valid_; }
 
 double Trajectory::getDuration() const { return trajectory_.back().time_; }
 
-bool Trajectory::assignData(TrajectoryContainer& trajectory, double path_tolerance) const
+bool Trajectory::assignData(TrajectoryContainer& trajectory) const
 {
-  auto it = trajectory_.begin();
+  Eigen::VectorXd path_lengths = Eigen::VectorXd::Zero(trajectory.size());
+  double length = 0;
+  for (Eigen::Index i = 1; i < trajectory.size(); ++i)
```

Comment:
Considering there are several loops here, can we get more descriptive indices? There are several i's here. I'm surprised clang-tidy isn't complaining

Reply:
I broke this out into separate function which should solve this .